contract_partner/models/contract.py

```diff
@@ -0,0 +1,24 @@
+# -*- coding: utf-8 -*-
+from openerp import models, fields, api
+
+
+class AccountAnalyticAccount(models.Model):
+    _inherit = 'account.analytic.account'
+
+    service_partner_id = fields.Many2one(
+        'res.partner',
+        string="Service Address")
+
+    contact_partner_id = fields.Many2one(
+        'res.partner',
+        string="Contact Address")
+
+    @api.model
+    def _prepare_invoice_data(self, contract):
+        invoice_vals = super(AccountAnalyticAccount, self).\
+            _prepare_invoice_data(contract)
+        if contract.service_partner_id:
+            invoice_vals['service_partner_id'] = contract.service_partner_id.id
+        if contract.contact_partner_id:
+            invoice_vals['contact_partner_id'] = contract.contact_partner_id.id
```

Comment:
Don't Invoices already have a field with the person/address where the Invoice should be sent?